src/main/java/impl/pair/NumberPairFactory.java

```diff
@@ -4,6 +4,6 @@
 
 public class NumberPairFactory {
     public static <K extends Number, T extends Number> Pair<K, T> of(K first, T second) {
-        return null;
+        return new PairImpl<>(first, second);
```

Comment:
There should be returned instance of NumberPair interface implementation

Reply:
But it umm is? 
`NumberPair` is `Pair` with limitations of type paratemers: `K extends Numner, T extends Number` 
We satisfy all conditions here since factory method limits type parameters. 

Hmmmm only `instanceof` would fail (had to doublecheck if that works with interfaces) 

Well, it's a 1min fix (it took more to write this comment =\)